add: Новый револьвер СБ, вариант к ваучерным пушкам

vovakons opened the pull request:
## Что этот ПР делает

Добавление нового револьвера для офицеров СБ. Можно получить как четвертый вариант пушки в ваучере.

Характеристики:
- Калибр: .45 Colt
- Емкость барабана: 6 патронов
- Возможность крепить модули: все модули доступные для энфорсеру
- Точность: пистолетная (*такая же как у энфорсера*)
- Отдача: средняя (*как у винтовок*)
- Скины: один бесплатный (*классический*), деревянная рукоятка за тир 1, полностью черный за тир 2.

Боеприпасы .45 Colt:
- Обычные: 28 урона
- Резина: 5 урона и 33 стамины
- Экспансивные: 35 урона и -50 бронепробоя
- Бронебойные: 20 урона и 30 бронепробития

**Дополнительно:** выбор кита через ваучер сб сделан через радиальное меню с показом пушки которую выбираешь.

## Почему это хорошо для игры

Больше пушек всегда хорошо, больше вариативности геймплея для сб.
Полный урон с барабана обычными пулями: 6*28=168 против энфорсера 23*8=184, плюс неудобная перезарядка.

Ну и потихоньку вводим донат скины в игру.

## Демонстрация изменений

<details><summary>Демонстрации изменений</summary>
<p>

Сам револьвер Таурус:
<img width="189" height="178" alt="image" src="https://github.com/user-attachments/assets/18625cf6-e104-4405-ab3e-22aefa40c003" />

Коробки с патронами для нового калибра .45 Colt
<img width="168" height="150" alt="image" src="https://github.com/user-attachments/assets/13494868-5ee0-4e33-91bb-66109638fd3d" />

Новые боксы стартовых китов для СБ: 
<img width="192" height="182" alt="image" src="https://github.com/user-attachments/assets/db2fd00f-a853-4b90-9815-a1dc5c5821ec" />


</p>
</details>

## Тестирование

Локалка.

vovakons committed:
taurum 9mm security revolver

Добавлен новый револьвер под 9 мм патроны для сб.
Модификатор урона 120% от калибра 9 мм.
Возможность крепить все типы пистолетных модулей.

vovakons committed:
taurus kit

Добавлен набор револьвера таурус в ваучер сб.
Добавлен спидлоадер для тауруса.

vovakons committed:
speedloader and kit box

Добавлены спрайты для спидлоадера и коробки с набором тауруса.
Добавлен рецепт пустого спидлоадера 9 мм в автолат сб.

vovakons committed:
Merge branch 'master220' into sec-revolver

vovakons committed:
merge fixes

vovakons commented:
Готов подумать над другим названием и изменением калибра если надо.

svotnik commented:
мог бы и скриншот сделать, что-бы оценить спрайт, мдэ

vovakons commented:
> мог бы и скриншот сделать, что-бы оценить спрайт, мдэ

смотри в демонстрацию изменений, там внутри спойлера

shahribek commented:
> Готов подумать над другим названием и изменением калибра если надо.

9мм револьверы существуют так что можно и не трогать.

vovakons commented:
> > Готов подумать над другим названием и изменением калибра если надо.
> 
> 9мм револьверы существуют так что можно и не трогать.

ну я в плане баланса, если эшель не понравится, на всякий случай

vovakons committed:
Merge branch 'master220' into sec-revolver

vovakons committed:
taurus revolver caliber to .45 colt

Изменение калибра револьвера Таурус на .45 Colt.
Добавлены коробки с новым калибром: обычные, резина, экспансивные и
бронебойные.

vovakons committed:
security voucher with radial menu

Использование радиального меню при выбора набора сб при использовании
ваучера.

vovakons commented:
Поменял калибр на новый .45 Colt

rafnik03 commented:
СБ набухнет от количества пушек

PiroMage commented:
> > Готов подумать над другим названием и изменением калибра если надо.
> 
> 9мм револьверы существуют так что можно и не трогать.

Ну он просто хотел сделать подвиды патронов для ревика, а если сделать ХП 9мм с 35 урона, то это можно будет запихнуть в A9 или форсер, что ну... Сделает это очень сильным...

vovakons commented:
> Ну он просто хотел сделать подвиды патронов для ревика, а если сделать ХП 9мм с 35 урона, то это можно будет запихнуть в A9 или форсер, что ну... Сделает это очень сильным...

ну я изначально хотел его под 9 мм сделать, но главный сказал сделать нвоый калибр, и чтобы новый калибр не был слишком бледный по сравнению с 9 мм сделал дополнительно ap и hp варианты

PiroMage commented:
> > Ну он просто хотел сделать подвиды патронов для ревика, а если сделать ХП 9мм с 35 урона, то это можно будет запихнуть в A9 или форсер, что ну... Сделает это очень сильным...
> 
> ну я изначально хотел его под 9 мм сделать, но главный сказал сделать нвоый калибр, и чтобы новый калибр не был слишком бледный по сравнению с 9 мм сделал дополнительно ap и hp варианты

Да это-то все хорошо получилось, просто думай наперед сколько нытья будет если смогут засунуть в форсер на 12-13 патронов эти ХПшки которые наносят 35 урона по мясу, или в А9 который будет просто сбривать голых антагов очередями. Поэтому имхо все круто и не надо делать 9мм ХП, а вот АП в теории можно было бы добавить, но это уже дело другого ПРа какого-нибудь.

vovakons commented:
> сто думай наперед сколько нытья будет если смогут засунуть в форсер на 12-13 патронов эти ХПшки которые наносят 35 урона по мясу, или в А9 который будет просто сбривать голых антагов очередями. Поэтому имхо все круто и не надо делать 9мм ХП, а вот АП в теории можно было бы добавить, но это

ну так изначально при 9 мм и не планировалось хп или ап патронов. Это добавилось после смены калибра, на 9 мм итак есть дизейблер и лазер патроны. А так 9 мм ап уже есть в коде, где то я видал

NuSanya commented:
в протолат бы экспансивки и бронепробивающие

vovakons commented:
> в протолат бы экспансивки и бронепробивающие

да они не такие мощные чтобы опять запереть за ебучую стойку рнд которая никак не играется

vovakons committed:
update sprites from lussat

oi98 commented:
Прикольные спрайты

PlayerUnknown14 commented on code/datums/skins/donat_skins.dm:
```diff
@@ -234,6 +234,25 @@
+	item_path = /obj/item/gun/projectile/revolver/taurus
+
+/datum/item_skin_data/taurus_revolver/classic
+	name = "Классическая"
```
```suggestion
	name = "Стандартный"
```

PlayerUnknown14 commented on code/datums/skins/donat_skins.dm:
```diff
@@ -234,6 +234,25 @@
+	donation_tier = 1
+
+/datum/item_skin_data/taurus_revolver/wood
+	name = "Деревянная рукоятка"
```
```suggestion
	name = "Классика"
```

PlayerUnknown14 commented on code/datums/skins/donat_skins.dm:
```diff
@@ -234,6 +234,25 @@
+	donation_tier = 1
+
+/datum/item_skin_data/taurus_revolver/black
+	name = "Чёрная"
```
```suggestion
	name = "Тактический"
```

PlayerUnknown14 commented on code/modules/projectiles/ammunition/boxes.dm:
```diff
@@ -1045,6 +1117,31 @@
+		DATIVE = "устройству быстрой зарядки (.45 Colt)",
+		ACCUSATIVE = "устройство быстрой зарядки (.45 Colt)",
+		INSTRUMENTAL = "устройством быстрой зарядки (.45 Colt)",
+		PREPOSITIONAL = "устройстве быстрой зарядки (.45 Colt)",
```
ускоритель заряжания

PlayerUnknown14 commented on code/modules/projectiles/ammunition/boxes.dm:
```diff
@@ -1045,6 +1117,31 @@
 
+/obj/item/ammo_box/speedloader/rubber45colt
+	name = "speed loader (.45 Colt)"
+	desc = "Устройство для быстрой зарядки револьверов патронами калибра .45 Colt."
```
```suggestion
	desc = "Устройство для быстрой перезарядки револьверов патронами калибра .45 Colt."
```

PlayerUnknown14 commented on code/modules/projectiles/ammunition/boxes.dm:
Ру_неймы вида "...ый .45 Colt", в единственном числе.

PlayerUnknown14 commented on code/modules/research/designs/autolathe_designs.dm:
Нужно поправить после локализации НИО.

PlayerUnknown14 commented on code/modules/projectiles/guns/projectile/revolver.dm:
```diff
@@ -152,6 +152,37 @@
+//Security revolver
+/obj/item/gun/projectile/revolver/taurus
+	name = "Taurus revolver"
+	desc = "Стандартный револьвер службы безопасности Нанотрейзен под калибр .45 Colt."
```
`"Револьвер под калибр .45 Colt, используемый силовыми структурами \"Нанотрейзен\". Отличается простотой конструкции, высокой надёжностью и минимальным количеством движущихся частей. Произведён \"Оружейной Ауссек\"."`

PlayerUnknown14 commented on code/modules/projectiles/guns/projectile/revolver.dm:
```diff
@@ -152,6 +152,37 @@
+		DATIVE = "Револьверу \"Таурус\"",
+		ACCUSATIVE = "Револьверу \"Таурус\"",
+		INSTRUMENTAL = "Револьвером \"Таурус\"",
+		PREPOSITIONAL = "Револьвере \"Таурус\"",
```
Со строчной.

PlayerUnknown14 commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -1040,6 +1040,27 @@
+		DATIVE = "набору Таурус",
+		ACCUSATIVE = "набор Таурус",
+		INSTRUMENTAL = "набором Таурус",
+		PREPOSITIONAL = "наборе Таурус",
```
`револьвера \"Таурус\"`

PlayerUnknown14 commented on code/game/objects/items/weapons/storage/boxes.dm:
```diff
@@ -1040,6 +1040,27 @@
 
+/obj/item/storage/box/taurus_kit
+	name = "taurus revolver kit (rubber)"
+	desc = "A box marked with pictures of an taurus revolver, two ammo clips, and the word 'NON-LETHAL'."
```
?

vovakons committed:
review fixes

vovakons committed:
Merge branch 'master220' into sec-revolver

vovakons committed:
review fixes 2

PlayerUnknown14 commented on code/modules/projectiles/guns/projectile/revolver.dm:
```diff
@@ -152,6 +152,37 @@
+//Security revolver
+/obj/item/gun/projectile/revolver/taurus
+	name = "Taurus revolver"
+	desc = "Револьвер под калибр .45 Colt, используемый силовыми структурами \"Нанотрейзен\". Отличается простотой конструкции, высокой надёжностью и минимальным количеством движущихся частей. Произведён \"Оружейной Ауссек\"."
```
Слеши и табуляция.

vovakons committed:
taurus description fix

LussatTheVulpa commented:
Я являюсь автором спрайтов коробок в следующих файлах:

icons/obj/storage.dmi

icons/obj/weapons/ammo.dmi

Данные работы были добавлены в этот репозиторий в коммите https://github.com/ss220-space/Paradise/pull/8035/changes/7ac7ed7452f8fb74c947b952f33c3bc000082b83 пользователем vovakons.

Я не давал письменного или явного согласия на лицензирование моих работ под лицензией CC BY-SA 3.0 данного репозитория. Моё неформальное разрешение на использование не распространялось на их публичное распространение на этих условиях.

Я официально требую удаления моих работ в течение 7 дней.

В случае если материалы не будут удалены, я буду вынужден подать уведомление о нарушении авторских прав по закону DMCA в GitHub.

LussatTheVulpa.

captainnelly commented:
> Я являюсь автором спрайтов коробок в следующих файлах:
> 
> icons/obj/storage.dmi
> 
> icons/obj/weapons/ammo.dmi
> 
> Данные работы были добавлены в этот репозиторий в коммите [7ac7ed7](https://github.com/ss220-space/Paradise/commit/7ac7ed7452f8fb74c947b952f33c3bc000082b83) пользователем vovakons.
> 
> Я не давал письменного или явного согласия на лицензирование моих работ под лицензией CC BY-SA 3.0 данного репозитория. Моё неформальное разрешение на использование не распространялось на их публичное распространение на этих условиях.
> 
> Я официально требую удаления моих работ в течение 7 дней.
> 
> В случае если материалы не будут удалены, я буду вынужден подать уведомление о нарушении авторских прав по закону DMCA в GitHub.
> 
> LussatTheVulpa.

Во первых ты тупой
Во вторых ты ведешь себя как обиженный долбоеб
В третьих это нихуя не меняет и лучше никому не делает
В четвертых тебя пошлет нахуй гитхаб и ты соснешь жирный вонючий пенис

LussatTheVulpa commented:
> > Я являюсь автором спрайтов коробок в следующих файлах:
> > icons/obj/storage.dmi
> > icons/obj/weapons/ammo.dmi
> > Данные работы были добавлены в этот репозиторий в коммите [7ac7ed7](https://github.com/ss220-space/Paradise/commit/7ac7ed7452f8fb74c947b952f33c3bc000082b83) пользователем vovakons.
> > Я не давал письменного или явного согласия на лицензирование моих работ под лицензией CC BY-SA 3.0 данного репозитория. Моё неформальное разрешение на использование не распространялось на их публичное распространение на этих условиях.
> > Я официально требую удаления моих работ в течение 7 дней.
> > В случае если материалы не будут удалены, я буду вынужден подать уведомление о нарушении авторских прав по закону DMCA в GitHub.
> > LussatTheVulpa.
> 
> Во первых ты тупой Во вторых ты ведешь себя как обиженный долбоеб В третьих это нихуя не меняет и лучше никому не делает В четвертых тебя пошлет нахуй гитхаб и ты соснешь жирный вонючий пенис

нет нелли, я делаю всё максимально правильно потому-что я красавчик и могу это сделать.

tsoivecen199015 commented:
> > Я являюсь автором спрайтов коробок в следующих файлах:
> > icons/obj/storage.dmi
> > icons/obj/weapons/ammo.dmi
> > Данные работы были добавлены в этот репозиторий в коммите [7ac7ed7](https://github.com/ss220-space/Paradise/commit/7ac7ed7452f8fb74c947b952f33c3bc000082b83) пользователем vovakons.
> > Я не давал письменного или явного согласия на лицензирование моих работ под лицензией CC BY-SA 3.0 данного репозитория. Моё неформальное разрешение на использование не распространялось на их публичное распространение на этих условиях.
> > Я официально требую удаления моих работ в течение 7 дней.
> > В случае если материалы не будут удалены, я буду вынужден подать уведомление о нарушении авторских прав по закону DMCA в GitHub.
> > LussatTheVulpa.
> 
> Во первых ты тупой Во вторых ты ведешь себя как обиженный долбоеб В третьих это нихуя не меняет и лучше никому не делает В четвертых тебя пошлет нахуй гитхаб и ты соснешь жирный вонючий пенис

Мув стрёмный, но это теоретическое описание случайной ошибки, бага, или утечки данных и прочее прочее. Ну то есть представь что вот прям щас произошла проблема что нужно срочно решить.

captainnelly commented:
> > > Я являюсь автором спрайтов коробок в следующих файлах:
> > > icons/obj/storage.dmi
> > > icons/obj/weapons/ammo.dmi
> > > Данные работы были добавлены в этот репозиторий в коммите [7ac7ed7](https://github.com/ss220-space/Paradise/commit/7ac7ed7452f8fb74c947b952f33c3bc000082b83) пользователем vovakons.
> > > Я не давал письменного или явного согласия на лицензирование моих работ под лицензией CC BY-SA 3.0 данного репозитория. Моё неформальное разрешение на использование не распространялось на их публичное распространение на этих условиях.
> > > Я официально требую удаления моих работ в течение 7 дней.
> > > В случае если материалы не будут удалены, я буду вынужден подать уведомление о нарушении авторских прав по закону DMCA в GitHub.
> > > LussatTheVulpa.
> > 
> > 
> > Во первых ты тупой Во вторых ты ведешь себя как обиженный долбоеб В третьих это нихуя не меняет и лучше никому не делает В четвертых тебя пошлет нахуй гитхаб и ты соснешь жирный вонючий пенис
> 
> нет нелли, я делаю всё максимально правильно потому-что я красавчик и могу это сделать.

Не по понятиям делаешь хуйню кринжовую
Ты бегал по дису чтобы твои хуйнюшки добавили в игрушки и теперь какое-то говно делаешь ради непонятно чего

tsoivecen199015 commented:
> > > > Я являюсь автором спрайтов коробок в следующих файлах:
> > > > icons/obj/storage.dmi
> > > > icons/obj/weapons/ammo.dmi
> > > > Данные работы были добавлены в этот репозиторий в коммите [7ac7ed7](https://github.com/ss220-space/Paradise/commit/7ac7ed7452f8fb74c947b952f33c3bc000082b83) пользователем vovakons.
> > > > Я не давал письменного или явного согласия на лицензирование моих работ под лицензией CC BY-SA 3.0 данного репозитория. Моё неформальное разрешение на использование не распространялось на их публичное распространение на этих условиях.
> > > > Я официально требую удаления моих работ в течение 7 дней.
> > > > В случае если материалы не будут удалены, я буду вынужден подать уведомление о нарушении авторских прав по закону DMCA в GitHub.
> > > > LussatTheVulpa.
> > > 
> > > 
> > > Во первых ты тупой Во вторых ты ведешь себя как обиженный долбоеб В третьих это нихуя не меняет и лучше никому не делает В четвертых тебя пошлет нахуй гитхаб и ты соснешь жирный вонючий пенис
> > 
> > 
> > нет нелли, я делаю всё максимально правильно потому-что я красавчик и могу это сделать.
> 
> Не по понятиям делаешь хуйню кринжовую Ты бегал по дису чтобы твои хуйнюшки добавили в игрушки и теперь какое-то говно делаешь ради непонятно чего

Погоди, а ты следил за политикой в последнее время?

vovakons commented:
Я не буду ничего откатывать, ебитесь сами и не срите мне уведомлениями гита

UnderwaterNeike commented:
Вован, спасибо за детство, легенда.🐣
